### PR TITLE
NCNE : Enable publish button only valid chart numbers

### PR DESCRIPTION
**Pull request by ssrajan**
1. Disable the publish button by default.
2. Enable it when metadata is retrieved from CARIS successfully
3. Disable it for Invalid chart numbers
4. Disable it again when the publish screen shown after publish errors.

### CODE REVIEW

File: src/NCNEPortal.UnitTests/PageValidationTests.cs [hunk 1]
```diff
@@ -29,9 +27,9 @@ public void Setup()
                 .UseInMemoryDatabase(databaseName: "inmemory")
                 .Options;
 
-            _dbContext = new NcneWorkflowDbContext(dbContextOptions);
+            new NcneWorkflowDbContext(dbContextOptions);
```
Comment (Rossall82) [inline src/NCNEPortal.UnitTests/PageValidationTests.cs:30]: What's going on here then?

File: src/NCNEPortal.UnitTests/PageValidationTests.cs [hunk 2]
```diff
@@ -29,9 +27,9 @@ public void Setup()
...
 
-            _fakeAdDirectoryService = A.Fake<IAdDirectoryService>();
+            A.Fake<IAdDirectoryService>();
```
Comment (Rossall82) [inline src/NCNEPortal.UnitTests/PageValidationTests.cs:32]: No assignment for this?

File: src/NCNEPortal.UnitTests/PageValidationTests.cs (hunk 1 above)
Comment (ssrajan) [reply src/NCNEPortal.UnitTests/PageValidationTests.cs]: removed the statement now

File: src/NCNEPortal.UnitTests/PageValidationTests.cs (hunk 2 above)
Comment (ssrajan) [reply src/NCNEPortal.UnitTests/PageValidationTests.cs]: removed the statement now

**Pull request merged**